File: proiectcolectiv/src/main/java/com/proiectcolectiv/models/document/Document.java
package com.proiectcolectiv.models.document;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.Entity;

/**
 * Created by achy_ on 1/1/2017.
 */


@NoArgsConstructor
@Getter
@Setter
@Entity
@ToString
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Document extends DocumentMetaData {

    public Document(Long id, String version, String author, String creationDate, String abstractText,
                    String keywords, String lastEditedOn, String lastEditedBy) {
        super(id, version, author, creationDate, abstractText, keywords, lastEditedOn, lastEditedBy);
    }

    public Document(Long id, String version, String author, String creationDate, String abstractText,
                    String keywords, String lastEditedOn, String lastEditedBy, String name,
                    String details, String documentType, DocumentStatus status) {
        super(id, version, author, creationDate, abstractText, keywords, lastEditedOn, lastEditedBy);
        this.name = name;
        this.details = details;
        this.documentType = documentType;
        this.status = status;
    }

    public Document(String version, String author, String creationDate, String abstractText,
                    String keywords, String lastEditedOn, String lastEditedBy, String name,
                    String details, String documentType, DocumentStatus status) {
        super(version, author, creationDate, abstractText, keywords, lastEditedOn, lastEditedBy);
        this.name = name;
        this.details = details;
        this.documentType = documentType;
        this.status = status;
    }

    private String name;

    private String details;

    private String documentType;

    private DocumentStatus status;


}

File: proiect-colectiv-angular2/app/documents/document.ts
export class Document {


  public id: number;
  public version: string;
  public author: string;
  public creationDate: string;
  public abstractText: string;
  public keywords: string;
  public lastEditedOn: string;
  public lastEditedBy: string;
  public name: string;
  public details: string;
  public documentType: string;
  public status: string;
  constructor(id: number, version: string, author: string, creationDate: string,
    abstractText: string, keywords: string, lastEditedOn: string, lastEditedBy: string,
    name: string, details: string, documentType: string, status: string) {
    this.id = id;
    this.version = version;
    this.author = author;
    this.creationDate = creationDate;
    this.abstractText = abstractText;
    this.keywords = keywords;
    this.lastEditedOn = lastEditedOn;
    this.lastEditedBy = lastEditedBy;
    this.name = name;
    this.details = details;
    this.documentType = documentType;
    this.status = status;
  }


}